File: 0753-cracking-the-safe/0753-cracking-the-safe.py
class Solution:
    def crackSafe(self, n: int, k: int) -> str:
        visited = set()
        ans = '0' * (n - 1)
        for i in range(k ** n):
            temp = ans[-n+1:] if n > 1 else ''
            for j in range(k - 1, -1, -1):
                if temp + str(j) not in visited:
                    visited.add(temp + str(j))
                    ans += str(j)
                    break
        return ans 

File: 0753-cracking-the-safe/0753-cracking-the-safe.js
/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var crackSafe = function(n, k) {
  const visited = new Set();
  let ans = '0'.repeat(n - 1);
  
  for (let i = 0; i < k ** n; i++) {
    let temp = ans.slice(ans.length - n + 1);
    for (let j = k - 1; j >= 0; j--) {
      if (!visited.has(temp + j)) {
        visited.add(temp + j);
        ans += j;
        break;
      }
    }
  }
  
  return ans;
};